libsigner/src/messages.rs

```diff
@@ -403,7 +404,12 @@ impl StacksMessageCodec for SignerMessage {
                 }
             }
             SignerMessageTypePrefix::EncryptedSignerState => {
-                let encrypted_state = read_next::<_, _>(fd)?;
+                // Typically the size of the signer state is much smaller, but in the fully degenerate case the size of dkg shares is
+                // (2800 + 2800) * 32 (threshold * shares * 32).
+                // The signer state holds some additional information, so we're adding a factor 4 to have some margin
```

Comment:
Is `* 4` enough margin?

Reply:
Yes, but looking at the saved state again it seems to be a bit more narrow than intended. As already established we have a polynomial of $t*32$ where $t=2800$ is the threshold, and key shares that gives another $t*32$ in the worst case. However, in addition to the shares we store their ID as a u32 twice (once with the shares in `PartyState` and once alone in `SignerState`) giving an additional term of $t*32*2$. This means that our state is of size $t*32*4 + C$ for some small constant $C$ (accounting for some additional fields in the `SignerState`, length prefixes etc.). This is slightly more than half the current margin.

While it is very unrealistic that we'd in practice get even close to this margin, I'd personally feel more comfortable with a larger margin.

What's the main purpose of using a bound reader here? Is it a security measure to prevent from certain dos attacks? In that case I suppose we could extend this margin to be much larger.

Reply:
Extended the margin to $16\cdot32t$.